app/controllers/concerns/metrics_queries.rb

```diff
@@ -0,0 +1,48 @@
+require "active_support/concern"
+
+module MetricsQueries
+  extend ActiveSupport::Concern
+  include MetricsUtils
+
+  def downloads_date_range_query(model, date_start, date_end, interval)
+    model_column, model_id = set_model_vars(model)
+
+    Rollups::HourlyDownload
+      .where("#{model_column}": model_id, hour: (date_start..date_end))
+      .select(model_column.to_s, "DATE_TRUNC('#{interval}', hour) AS hour", "SUM(count) AS count")
+      .group(model_column.to_s, "DATE_TRUNC('#{interval}', hour) AS hour")
+      .order(Arel.sql("DATE_TRUNC('#{interval}', hour) ASC"))
```

Comment:
Just noting, you'll have to add `.final` (the [FINAL](https://clickhouse.com/docs/sql-reference/statements/select/from#final-modifier) modifier) to all these queries, now that updates are coming in throughout the day.